web/client/components/dashboard/ViewSwitcher.jsx

```diff
@@ -0,0 +1,158 @@
+import React, { useRef, useState } from 'react';
+import { Button, Dropdown, Glyphicon, MenuItem } from 'react-bootstrap';
+import Message from '../I18N/Message';
+import withConfirm from '../misc/withConfirm';
+import FlexBox from '../layout/FlexBox';
+import useCheckScroll from './hooks/useCheckScroll';
+
+const WithConfirmButton = withConfirm(MenuItem);
+
+const View = ({ handleSelect, isSelected, id, color, name, onRemove, onMove, canDelete, onConfigure, canMoveLeft, canMoveRight, canEdit }) => {
+    const [position, setPosition] = useState({ left: 0, bottom: 0 });
+    const toggleBtnRef = useRef(null);
+
+    const handleToggleClick = () => {
+        handleSelect(id);
+        if (toggleBtnRef.current) {
+            const rect = toggleBtnRef.current.getBoundingClientRect();
+            setPosition({
+                left: rect.left,
+                bottom: window.innerHeight - rect.top
+            });
+        }
+    };
+
+    return (
+        <Dropdown
+            dropup
+            bsStyle="default"
+            id="split-button-dropup-pull-right"
+            className={`${isSelected ? "is-selected " : ""}layout-views _relative`}
+            style={{ borderBottom: `2px solid ${color}` }}
+        >
+            <Button
+                onClick={() => handleSelect(id)}
+                bsStyle="default" className="layout-views-dropdown _padding-tb-sm"
+            >
+                {name}
+            </Button>
+            {canEdit && (
+                <Dropdown.Toggle
+                    onClick={handleToggleClick}
+                    noCaret
+                    className="square-button-md dashboard-view-actions _relative _padding-tb-sm _padding-r-sm"
+                >
+                    <div ref={toggleBtnRef}>
+                        <Glyphicon glyph="option-vertical" />
+                    </div>
+                </Dropdown.Toggle>
+            )}
+            <Dropdown.Menu className="_fixed" style={{ left: `${position.left}px`, bottom: `${position.bottom}px` }}>
```

Comment:
I think we should find a way to pull the menu with right alignement when crossing the half with of the page

<img width="509" height="217" alt="image" src="https://github.com/user-attachments/assets/705e31a6-70f2-4a96-a60f-9a612c742039" />